frontend/src/Pages/FC/bans/Modals.js

```diff
@@ -0,0 +1,235 @@
+  const toastContext = React.useContext(ToastContext);
+
+  const onClick = () => {
+    if (pending) {
+      return; // stop users from clicking this twice
+    }
+    isPending(true);
+
+    errorToaster(
+      toastContext,
+      apiCall(`/api/v2/bans/${ban.id}`, {
+        method: "DELETE",
+      })
+        .then(() => {
+          isPending(false);
+          setOpen(false);
+          refreshFunction();
+        })
+        .catch((err) => {
+          isPending(false);
+          throw err;
+        })
+    );
+  };
+
+  return (
+    <Modal open={isOpen} setOpen={setOpen}>
+      <Box>
+        <Title>Revoke {ban?.entity.category} Ban?</Title>
+
+        <div style={{ maxWidth: "500px", display: "flex" }}>
+          <FormGroup>
+            <Label>{ban?.entity.category}:</Label>
+            <EntityLink {...ban} />
+          </FormGroup>
+
+          <FormGroup>
+            <Label>Issued By:</Label>
+            <CharacterName {...ban.issued_by} />
+          </FormGroup>
+        </div>
+
+        <FormGroup>
+          <Label>Reason:</Label>
+          <p style={{ maxWidth: "500px" }}>{ban.reason}</p>
+        </FormGroup>
+
+        <CenteredButtons size={"90px"}>
+          <Button variant="secondary" onClick={() => setOpen(false)}>
+            Cancel
+          </Button>
+          <Button variant="danger" onClick={onClick} disabled={pending}>
+            Confirm
+          </Button>
+        </CenteredButtons>
+      </Box>
+    </Modal>
+  );
+};
+
+const Update = ({ ban, isOpen, setOpen, refreshFunction }) => {
+  const [revoked_at, setRevokedAt] = React.useState(undefined);
+  const [reason, setReason] = React.useState(undefined);
+  const [public_reason, setPublicReason] = React.useState(undefined);
+  const [isPermanent, setPermanent] = React.useState(false);
+  const [pending, isPending] = React.useState(false);
+  const toastContext = React.useContext(ToastContext);
+
+  const onClick = (e) => {
+    e.preventDefault();
+
+    if (pending) {
+      return; // stop users from clicking this twice
+    }
+    isPending(true);
+
+    errorToaster(
+      toastContext,
+      apiCall(`/api/v2/bans/${ban.id}`, {
+        method: "PATCH",
+        json: {
+          reason,
+          public_reason,
+          revoked_at: isPermanent ? null : revoked_at,
+        },
+      })
+        .then(() => {
+          isPending(false);
+          setOpen(false);
+          refreshFunction();
+        })
+        .catch((err) => {
+          isPending(false);
+          throw err;
+        })
+    );
+  };
+
+  useEffect(() => {
+    if (revoked_at !== ban?.revoked_at) {
+      setRevokedAt(revoked_at);
+      if (!revoked_at) {
+        setPermanent(true);
+      }
+    }
+    if (reason !== ban?.reason) {
+      setReason(reason);
+    }
+    if (public_reason !== ban?.public_reason) {
+      setPublicReason(public_reason);
+    }
+  }, [ban?.public_reason, ban?.reason, ban?.revoked_at, public_reason, reason, revoked_at]);
```

Comment:
I need to double-check this logic as it is continuing to cause problems.  Currently, the modal is not updating when the `/api/v2/bans` endpoint is called.  This function should update state when changes come through props, while otherwise allowing the components to control themselves.